rustls/src/crypto/mod.rs

```diff
@@ -329,3 +335,41 @@ impl From<&[u8]> for SharedSecret {
         Self(source.to_vec())
     }
 }
+
+#[cfg(any(feature = "ring", feature = "fips"))]
+pub(crate) fn default_provider() -> CryptoProvider {
```

Comment:
So this brings back the situation where a change in Cargo features can change the crypto provider. That still seems undesirable to me?

I guess it's less bad maybe because it fails closed in the sense that if either the application or a library enables `fips`, this will always default to `fips`, but if my application is set up to use `ring` and a library forces a change to `fips` that still seems suboptimal to me.

Reply:
I think it _is_ desirable, in this case. Someone asking for the `fips` feature anywhere in the dependency tree should alter the default for every caller not being more specific, because I think the goal of such a user is to eradicate all other crypto from their program.

But I think it's clear we need to reach an acceptable conclusion to #1744 and #1642.